src/pool/pool_disjoint.c

```diff
@@ -175,10 +191,10 @@ static void slab_free_chunk(slab_t *slab, void *ptr) {
     // Make sure that we're in the right slab
     assert(ptr >= slab_get(slab) && ptr < slab_get_end(slab));
 
-    // Even if the pointer p was previously aligned, it's still inside the
-    // corresponding chunk, so we get the correct index here.
-    size_t chunk_idx =
-        ((uintptr_t)ptr - (uintptr_t)slab->mem_ptr) / slab->bucket->size;
+    // Get the chunk index
+    uintptr_t ptr_diff = (uintptr_t)ptr - (uintptr_t)slab->mem_ptr;
+    assert(ptr_diff / slab->bucket->size == 0);
+    size_t chunk_idx = ptr_diff / slab->bucket->size;
```

Comment:
Something's not right, `ptr_diff / slab->bucket->size` is asserted to be `0` the line above, so the `chunk_idx` wll be `0` too.

Reply:
thanks, fixed